python-package/xgboost/data.py

```diff
@@ -1208,7 +1215,7 @@ def _proxy_transform(
         data, _ = _ensure_np_dtype(data, data.dtype)
         return data, None, feature_names, feature_types
     if _is_scipy_csr(data):
-        data = _transform_scipy_csr(data)
+        data = transform_scipy_sparse(data, True)
```

Comment:
So we don't support CSC for data iterator just yet?

Reply:
We can support it by transforming it into CSR. libxgboost sketching implementation (quantile dmatrix) works on row-major data. Column major input support will require non-trivial effort.

Reply:
Up to you.

Reply:
Opened an issue to track: https://github.com/dmlc/xgboost/issues/8740 .